lib/cli/run-serverless-tencent.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const { createRequire } = require('module');
+const os = require('os');
+const path = require('path');
+const fsp = require('fs').promises;
+const childProcess = require('child_process');
+const spawn = require('child-process-ext/spawn');
+
+const relativeBinPath = 'serverless-tencent/bin/serverless-tencent';
+const standaloneFilename = path.resolve(
+  os.homedir(),
+  `.serverless-tencent/bin/serverless-tencent${process.platform === 'win32' ? '.exe' : ''}`
+);
+const standaloneServerUrl = 'https://slt-binary-sv-1300963013.cos.accelerate.myqcloud.com';
+
+const resolveAbsoluteModulePath = (contextDirname, modulePath) => {
+  try {
+    return createRequire(contextDirname).resolve(modulePath);
+  } catch {
+    return null;
+  }
+};
+
+const resolveGlobalNpmPath = async () => {
+  const npmNodeModulesPath = await (async () => {
+    try {
+      return String((await spawn('npm', ['root', '-g'])).stdoutBuffer);
+    } catch (error) {
+      return null;
+    }
+  })();
+
+  if (!npmNodeModulesPath) return null;
+  try {
+    return require.resolve(`${npmNodeModulesPath}/${relativeBinPath}`);
+  } catch (globalDepError) {
+    return null;
+  }
+};
+
+const isStandaloneInstalled = async () => {
+  try {
+    await fsp.access(standaloneFilename);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+const resolvePlatform = () => {
+  switch (process.platform) {
+    case 'darwin':
+      return 'macos';
+    case 'win32':
+      return 'win';
+    default:
+      return process.platform;
+  }
+};
+const resolveArch = () => {
+  switch (process.arch) {
+    case 'x32':
+      return 'x86';
+    case 'arm':
+    case 'arm64':
```

Comment:
I believe we need to have the same logic as for standalone installation for M1 Macs here - based on conversation in related Asana ticket, there will be no dedicated native M1 binary but instead it will use x64 version with Rosetta emulation, could you confirm @zongUMR ?

Reply:
It's a good point. I've intentionally left it that way, as it's better if Tencent host x86 binary under two names, instead of falling back `x86` when in `arm`.

Reason is that Tencent users will rather not upgrade Framework CLI.

So if at some point M1 dedicated binary will be generated on Tencent side, it'll be easier to just upload it at right location, instead of updating CLI triage, of which update is unlikely to be picked up by existing users

Reply:
I see, that sounds like a good idea 👍

Reply:
Actually I have the same problem with @pgrzesik and I am not clear why we do not return a `x64` for a mac m1 users just like Piotr's this [PR](https://github.com/serverless/serverless/pull/10291/files#diff-dcb392acdb47ffe9877460c1bac64469954c1466b1f96e7573cd1b9155473ec6R24) , if a m1 user install the serverless-tencent binary, it will request the url: `xxxxx/serverless-tencent-macos-armv6`, but we only have a `serverless-tencent-macos-x64` and want it run for the m1 users?

Reply:
@zongUMR please see my comment above.

It's very unlikely that Tencent users, will update `serverless` installation, technically they do not have to, it serves only as entry door, and only triage logic from it it's executed. Also they will never be notified of new `serverless` releases.

So if now in triage we will decide to unconditionally serve `x64` to M1 users, it'll stay that for users even after you eventually start building dedicated binary for M1. It'll be difficult to push them to update `serverless` so triage logic is updated for them.

I think it'll be better that you instead now upload the build into two locations, as then when you find the way to build M1 dedicated binary, you can just provide it at ARM dedicated URL (where so far `x64` binary wsas provided). Having that  there will no changes needed on triage side.

Reply:
Still, what I proposed doesn't envision showing the warning about Rosetta to users.

If we want to show that warning, then indeed we need to add conditional here, but then there's a risk of M1 users sticking to `x64` build long after M1 dedicated binary is provided.